nservicebus/throughput-tool/postgresql-transport.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Measuring system throughput using PostgreSQL Transport
+summary: Use the Particular throughput tool to measure the throughput of an NServiceBus system.
+reviewed: 2024-05-22
+related:
+  - nservicebus/throughput-tool
+---
+
+The Particular throughput tool can be installed locally and run against a production system to discover the throughput of each endpoint in a system over a period of time.
+
+This article details how to collect endpoint and throughput data when the system uses the [PostgreSQL transport](/transports/postgresql/). Refer to the [throughput counter main page](./) for information how to install/uninstall the tool or for other data collection options.
+
+## Running the tool
+
+Once installed, execute the tool with the database connection string used by PostgreSQL endpoints.
+
+If the tool was [installed as a .NET tool](/nservicebus/throughput-tool/#installation-net-tool-recommended):
+
+```shell
+throughput-counter postgresql [options] --connectionString "Server=SERVER;Database=DATABASE;Port=5432;User Id=USERID;Password=PASSWORD;"
+```
+
+Or, if using the [self-contained executable](/nservicebus/throughput-tool/#installation-self-contained-executable):
+
+```shell
+Particular.EndpointThroughputCounter.exe postgresql [options] --connectionString "Server=SERVER;Database=DATABASE;Port=5432;User Id=USERID;Password=PASSWORD;"
+```
+
+The tool will run for slightly longer than 24 hours in order to capture a beginning and ending identity value for each queue table.
+
+## Options
+
+Either the `--connectionString` or `--connectionStringSource` must be used to provide the tool with connection string information.
+
+| Option | Description |
+|-|-|
+| <nobr>`--connectionString`</nobr> | A single database connection string<sup>1</sup> that will provide at least read access to all queue tables. |
+| <nobr>`--addCatalogs`</nobr> | When the `--connectionString` parameter points to a single database, but multiple database catalogs on the same server also contain NServiceBus message queues, the `--addCatalogs` parameter specifies additional database catalogs to search. The tool replaces the `Database` parameter in the connection string with the additional catalog and queries all of them. With this option, only a single database server is supported.<br/><br/>Example: `--connectionString <Catalog1String> --addCatalogs Catalog2 Catalog3 Catalog4` |
```

Comment:
is the terminology correct here for postgres? Do they use "catalog"?

Reply:
Yeah I wasnt sure as Ive seen the term used by ppl talking about postgresql, but it could be because they come from sqlserver background. Googling it I come across system calatogs, which are different.
I will change it to use "database" instead